packages/ast-spec/src/base/Position.ts

```diff
@@ -1,4 +1,4 @@
-export interface LineAndColumnData {
+export interface Position {
```

Comment:
The reason that I didn't use `Position` was because there [used to be a global type called `Position` in TS's lib.dom types](https://www.typescriptlang.org/play?ts=4.0.5#code/CYUwxgNghgTiAEYD2A7AzgF3gDwFzwAUk0BLDE1AbngHob4Ul4QYYkZ4AjcKAVzQRFS5VM2wlMaeCRTwA5hCScoEeBgCeABxBogA). (Note: [`lib.dom` is included by default if you specify no libs](https://github.com/microsoft/TypeScript/blob/main/lib/lib.d.ts#L22).)
I kept on accidentally referencing the wrong type (/ forgetting to import) due to the name clash - hence I named it `LineAndColumnData`.

It looks like [the type was renamed in ~4.1 to `GeolocationPosition`](https://github.com/microsoft/TypeScript/commit/16031bc429305e5daabf263f208678a6729a161e#diff-796971dc74a3f3cd77ca0a3a776d6fe27ab224cd501d9652dbeffcf45ba0f058L11790-L11794) however, so this won't be a problem with "modern" TS versions.

`Position` is a very generic name though that will conflict with any number of other packages that could be installed - so I'm not *entirely* sure if we should update it to align with the ESTree docs (which nobody really reads 😅).

It's not a hard stance

Reply:
I thought bringing it more in line would help with referencing the spec whenever it needs to be worked on, so that's why I opened the PR